### ISSUE TITLE
check DOI is minted at point of publication

### ISSUE THREAD

**Issue by only1chunts** [gigascience/gigadb-website#1641]
# User story

>As a curator
>I want to know the Dataset DOI is live at the point of publication
>So that we dont have public datasets without a working DOI

# Acceptance criteria

>Given I am on a Dataset Admin page
>When I update the status to "Publish"
>Then there is a check made on the DataCite website or API that the dataset DOI has been minted

>Given I am on a Dataset Admin page
>When I update the status to "Publish"
>And there is a check made on the DataCite  API that the dataset DOI has not been minted
>Then I see a message "The DOI hasn't been minted yet. Please do it"
>And I see a link "Click here to mint" in the error message box

# Additional Information

Occassionally a dataset is set to publish status but the DOI has not been minted with DataCite. This could be due to human error (forgot to click the button) or sometimes the action of the button fails for some technical reason, like DataCite API being down or something, and the user doesn't notice.

It would be nice if the status can only be set to "Publish" if the DataCite DOI is valid and registered with DataCite. 

I can also see no reason why the action of setting the status to Publish shouldn't action the minting of the DOI!
 
If we did that, we would STILL need the manual mintDOI button as that gets used to make updates to the DataCite metadata as well, and we may have instances where we need to mint the DOI prior to setting the status to "publish". If the DOI for the dataset already has been minted then the button should be labelled as updateDOI or something similar.

**Comment by only1chunts** [comment]
Just a note to add additional information: DataCite have the ability to pre-assign DOI numbers now (this was not an option when we first started using DataCite). In the future we may want to pre-assign the DOI's with DataCite at a dataset status before Publish, maybe at the pending status (still needs some thought). I'm mentioning it here incase it makes any difference to how you implement things now, i.e. when looking in DataCite for a related DOI that DOI in DataCite will need to be of the correct type, i.e. not "[Draft](https://support.datacite.org/docs/doi-states#draft-record)".
see #251

**Comment by alli83** [comment]
So this means that up until this status, any DOI can be provided or cause an error and is accepted, except at the final status?

**Comment by only1chunts** [comment]
@alli83 sorry but I dont fully understand your statement so I cant properly answer it:
"up until this status" - which status?
"any DOI can be provided or cause an error and is accepted" - provided where? accepted by what/who?

**Comment by alli83** [comment]
@only1chunts sorry, I mean "published" status? and DOI field on dataset form because in order to save a dataset, this field is required.

**Comment by only1chunts** [comment]
correct, the DOI field on the dataset admin form must have a value (regardless of what the dataset status is).
The point of this ticket is to ensure that the status of a dataset cannot be saved as "publish" without the value of the DOI field representing a working (minted) datacite public link, i.e. the mint DOI button has already been clicked AND was successful in its action.
